utils/utils.go

```diff
@@ -119,3 +125,30 @@ func splitEnvVar(envVar string) (key, value string) {
 	}
 	return split[0], strings.Join(split[1:], "=")
 }
+
+// If an output dir was provided through --output-dir flag, we create in the provided path, a new directory with a new file containing the scan results
+// This functionallity is being used for Jas scanners and SCA scanner
+func DumpScanResultsToFileIfNeeded(results interface{}, scanResultsOutputDir string, scanType string) (err error) {
+	// TODO this function should be in utils/results/results.go after the refactor, since it is a common code for Jas and SCA scanners
+	// TODO AFTER merging the refactor - make sure to create a new directory for every Scan Target and convert results to Sarif before writing them to file
+	if scanResultsOutputDir == "" || reflect.ValueOf(results).IsNil() {
```

Comment:
```suggestion
	if scanResultsOutputDir == "" ||  results == nil {
```
Why would result be nil? I would avoid using `reflect`